opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -2189,12 +2189,41 @@ namespace Opm
     void
     StandardWell<TypeTag>::
     computeWellRatesWithBhp(const Simulator& ebosSimulator,
-                            const EvalWell& bhp,
+                            const double& bhp,
+                            const bool iterate,
                             std::vector<double>& well_flux,
-                            Opm::DeferredLogger& deferred_logger) const
+                            Opm::DeferredLogger& deferred_logger)
     {
+
+
         const int np = number_of_phases_;
         well_flux.resize(np, 0.0);
+        WellControls* wc = well_controls_;
+        const double orig_bhp = well_controls_iget_target(wc, BHP);
```

Comment:
I do not understand the usage `well_controls_iget_target(wc, BHP);`. 

Please excuse me if I misread it. I think it is a wrong usage.

Reply:
Based on your suggestion I changed the code such that the control is changed rather than explicitly setting the bhp in the equation. Here is just store the original bhp control target value so that I can reset the control later.

Reply:
I think `BHP` is a `enum` control mode. I do not think the usage of `well_controls_iget_target(wc, BHP)` is correct.

Reply:
You are right. I havn't been working the the well controls in awhile. I guess I will have to iterate over all controls and pick the one with BHP.